playground/rsc-vite-framework/start-vite-middleware.js

```diff
@@ -0,0 +1,42 @@
+import { createRequestListener } from "@mjackson/node-fetch-server";
+import express from "express";
+
+const viteDevServer =
+  process.env.NODE_ENV === "production"
+    ? undefined
+    : await import("vite").then(({ createServer }) =>
+        createServer({
+          server: {
+            middlewareMode: true,
+          },
+        }),
+      );
+
+const requestListener = viteDevServer
+  ? async (req, res) => {
+      // In dev mode, ensure we load a fresh request handler every request
+      const rscEntry = await viteDevServer.environments.rsc.runner.import(
+        "virtual:react-router/unstable_rsc/rsc-entry",
+      );
+      return createRequestListener(rscEntry.default)(req, res);
+    }
```

Comment:
During dev, currently this does nothing since rsc plugin already includes the same handler in `viteDevServer.middlewares`. I noticed that it isn't the case in non-rsc framework mode. Should rsc plugin align with it?

https://github.com/remix-run/react-router/blob/e495e0eef712794a422882eaa4d70ed2a1a22844/packages/react-router-dev/vite/plugin.ts#L1623-L1626

(Btw, I noticed there's nothing specific about react router for this custom server setup and I'm thinking to add a same example on rsc plugin https://github.com/vitejs/vite-plugin-react/pull/817)